### PR TITLE
IE7 Fix word count on load and paste

### PR DESCRIPTION
**Pull request by hammerfunk**
Removed the on load word count which wasn’t needed
Fixed the paste word count

### CODE REVIEW

File: app/assets/javascripts/frontend/text-character-count.js [hunk 1]
```diff
@@ -41,31 +41,31 @@ $.fn.charcount = function() {
     }
   });
 
+  // If character count is over the limit then show error
+  var characterOver = function(counter, textInput) {
+    var lastLetter = textInput.val()[textInput.val().length - 1];
+    var maxWordCount = parseInt(textInput.attr("data-word-max"));
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Missing radix parameter.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 2]
```diff
@@ -41,31 +41,31 @@ $.fn.charcount = function() {
...
+    var maxWordCountLimit = parseInt(textInput.attr("data-word-max-limit"));
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Missing radix parameter.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 3]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
   // Maxlength for pasting text
   this.bind("paste", function(e){
-    e.preventDefault();
-    var pasteThis = ((e.originalEvent || e).clipboardData.getData("text/plain"));
-
-    for (var c = 0; c<pasteThis.length; c++) {
-      if (((typeof($(this).attr("maxlength")) !== typeof(undefined)) && $(this).attr("maxlength") !== false) == false || $(this).val().length <= $(this).attr("maxlength")) {
-        $(this).val($(this).val() + pasteThis[c]);
-        Countable.once(this, counting);
+    var original_text = $(this).val() + ((e.originalEvent || e).clipboardData.getData("text/plain"));
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Line is too long.<br>Identifier 'original_text' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 4]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+
+    if (original_text.length > 0) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'original_text' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 5]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+      e.preventDefault();
+      var char_test = $(this).after("<p class='char-count-test'></p>");
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'char_test' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 6]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+      char_test.attr("data-word-max", $(this).attr("data-word-max"));
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'char_test' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 7]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+      char_test.attr("data-word-max-limit", $(this).attr("data-word-max-limit"));
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'char_test' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 8]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+      var last_char = 0;
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'last_char' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 9]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+      $(this).val("");
+      for (c = 0; c<original_text.length; c++) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'original_text' is not in camel case.<br>'c' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 10]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+        if (((typeof(char_test.attr("maxlength")) !== typeof(undefined)) && char_test.attr("maxlength") !== false) == false || c <= char_test.attr("maxlength")) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Line is too long.<br>Identifier 'char_test' is not in camel case.<br>Use '===' to compare with 'false'.<br>'c' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 11]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+          char_test.text(original_text.substr(0, c + 1));
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'char_test' is not in camel case.<br>Identifier 'original_text' is not in camel case.<br>'c' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 12]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+          Countable.once(char_test[0], counting);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'char_test' is not in camel case.<br>'Countable' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 13]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+        }
+        if (((typeof(char_test.attr("maxlength")) !== typeof(undefined)) && char_test.attr("maxlength") !== false) == false || c <= char_test.attr("maxlength")) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Line is too long.<br>Identifier 'char_test' is not in camel case.<br>Use '===' to compare with 'false'.<br>'c' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 14]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+          last_char = c;
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'last_char' is not in camel case.<br>'c' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 15]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+        }
       }
+      $(this).val(original_text.substr(0, last_char + 1));
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Identifier 'original_text' is not in camel case.<br>Identifier 'last_char' is not in camel case.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 16]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
+      Countable.once(this, counting);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'Countable' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 17]
```diff
@@ -73,37 +73,39 @@ $.fn.charcount = function() {
...
-    for (c = 0; c<loaded_text.length; c++) {
-      if (((typeof($(this).attr("maxlength")) !== typeof(undefined)) && $(this).attr("maxlength") !== false) == false || $(this).val().length <= $(this).attr("maxlength")) {
-        $(this).val($(this).val() + loaded_text[c]);
-        Countable.once(this, counting);
-      }
-    }
-
     // Makes word count dynamic
-    Countable.live(this, counting)
+    Countable.live(this, counting);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'Countable' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 18]
```diff
@@ -73,37 +73,40 @@ $.fn.charcount = function() {
...
+
+    if (originalText.length > 0) {
+      e.preventDefault();
+      var charTest = $(this).after("<p class='char-count-test'></p>");
+      charTest.attr("data-word-max", $(this).attr("data-word-max"));
+      charTest.attr("data-word-max-limit", $(this).attr("data-word-max-limit"));
+      var lastChar = 0;
+      $(this).val("");
+      for (var c = 0; c<originalText.length; c++) {
+        if (((typeof(charTest.attr("maxlength")) !== typeof(undefined)) && charTest.attr("maxlength") !== false) === false || c <= charTest.attr("maxlength")) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Line is too long.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 19]
```diff
@@ -73,37 +73,40 @@ $.fn.charcount = function() {
...
+          charTest.text(originalText.substr(0, c + 1));
+          Countable.once(charTest[0], counting);
+        }
+        if (((typeof(charTest.attr("maxlength")) !== typeof(undefined)) && charTest.attr("maxlength") !== false) === false || c <= charTest.attr("maxlength")) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: Line is too long.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 20]
```diff
@@ -73,37 +73,40 @@ $.fn.charcount = function() {
...
+      e.preventDefault();
+      var charTest = $(this).after("<p class='char-count-test'></p>");
+      charTest.attr("data-word-max", $(this).attr("data-word-max"));
+      charTest.attr("data-word-max-limit", $(this).attr("data-word-max-limit"));
+      var lastChar = 0;
+      $(this).val("");
+      for (var c = 0; c<originalText.length; c++) {
+        if (((typeof(charTest.attr("maxlength")) !== typeof(undefined)) && charTest.attr("maxlength") !== false) === false || c <= charTest.attr("maxlength")) {
+          charTest.text(originalText.substr(0, c + 1));
+          Countable.once(charTest[0], counting);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'Countable' is not defined.

File: app/assets/javascripts/frontend/text-character-count.js [hunk 21]
```diff
@@ -73,37 +73,40 @@ $.fn.charcount = function() {
...
+        }
+        if (((typeof(charTest.attr("maxlength")) !== typeof(undefined)) && charTest.attr("maxlength") !== false) === false || c <= charTest.attr("maxlength")) {
+          lastChar = c;
+        }
       }
+      $(this).val(originalText.substr(0, lastChar + 1));
+      Countable.once(this, counting);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'Countable' is not defined.

**Comment by RuslanHamidullin** [comment]
Lookks fine :+1:

File: app/assets/javascripts/frontend/text-character-count.js [hunk 22]
```diff
@@ -73,37 +73,29 @@ $.fn.charcount = function() {
...
+      e.preventDefault();
 
-    for (var c = 0; c<pasteThis.length; c++) {
-      if (((typeof($(this).attr("maxlength")) !== typeof(undefined)) && $(this).attr("maxlength") !== false) == false || $(this).val().length <= $(this).attr("maxlength")) {
-        $(this).val($(this).val() + pasteThis[c]);
-        Countable.once(this, counting);
+      for (var c = 0; c<originalText.length; c++) {
+        if (((typeof($(this).attr("maxlength")) !== typeof(undefined)) && $(this).attr("maxlength") !== false) == false || $(this).val().length <= $(this).attr("maxlength")) {
+          $(this).val($(this).val() + originalText[c]);
+          Countable.once(this, counting);
```
Comment (matthewford) [inline app/assets/javascripts/frontend/text-character-count.js]: 'Countable' is not defined.

File: app/assets/javascripts/frontend/textarea-autoResize.js [hunk 23]
```diff
@@ -44,7 +44,7 @@ $(function() {
     resizeTextarea($(this));
   });
 
-  txt.on("keyup", function (e) {
+  txt.on("keyup paste", function (e) {
```
Comment (matthewford) [inline app/assets/javascripts/frontend/textarea-autoResize.js]: Mixed double and single quotes.

**Pull request merged**